src/message.spec.ts

```diff
@@ -127,3 +127,52 @@ test('self()', t => {
+
+  const RAW_OBJ = JSON.parse(`{"Alias":"","AppAccountFlag":0,"AttrStatus":0,"ChatRoomId":0,"City":"","ContactFlag":2,"DisplayName":"","EncryChatRoomId":"@305fa7624e77bbe1857bfc899dc05835","HeadImgUrl":"/cgi-bin/mmwebwx-bin/webwxgetheadimg?seq=648035057&username=@@f47ccb7762cf4dc7352508fb03b79b44cedd3d1e16152e7f1bae3d4cf7996eb2&skey=","HideInputBarFlag":0,"IsOwner":1,"KeyWord":"","MMFromBatchGet":true,"MMFromBatchget":true,"MMInChatroom":true,"MMOrderSymbol":"FUFEIRUQUN","MemberCount":3,"MemberList":[{"AttrStatus":2147584103,"DisplayName":"","KeyWord":"qq5","MemberStatus":0,"NickName":"22acb030-ff09-11e6-8a73-cff62d9268c5","PYInitial":"","PYQuanPin":"","RemarkPYInitial":"","RemarkPYQuanPin":"","Uin":0,"UserName":"@c8e11ab44ea04b92c98885453eeef1a5"},{"AttrStatus":135205,"DisplayName":"","KeyWord":"","MemberStatus":0,"NickName":"小桔同学","PYInitial":"","PYQuanPin":"","RemarkPYInitial":"","RemarkPYQuanPin":"","Uin":0,"UserName":"@56c8d702f12e05ae64872e7c2a92073486b57623bf04941cef47bf2e8452ca7d"},{"AttrStatus":233509,"DisplayName":"","KeyWord":"","MemberStatus":0,"NickName":"50fb16c0-ff09-11e6-9fe5-dda97284d25b","PYInitial":"","PYQuanPin":"","RemarkPYInitial":"","RemarkPYQuanPin":"","Uin":0,"UserName":"@4731ed48f9914e7b95ebec77803223863e2cd33497f8633c755d985a544c6922"}],"NickName":"付费入群","OwnerUin":0,"PYInitial":"FFRQ","PYQuanPin":"fufeiruqun","Province":"","RemarkName":"","RemarkPYInitial":"","RemarkPYQuanPin":"","Sex":0,"Signature":"","SnsFlag":0,"StarFriend":0,"Statues":1,"Uin":0,"UniFriend":0,"UserName":"@@f47ccb7762cf4dc7352508fb03b79b44cedd3d1e16152e7f1bae3d4cf7996eb2","VerifyFlag":0,"stranger":true}`)
+
+  const CONTACT_LIST = JSON.parse(`{"@c8e11ab44ea04b92c98885453eeef1a5":{ "UserName": "@c8e11ab44ea04b92c98885453eeef1a5","NickName": "李佳芮","RemarkName": "22acb030-ff09-11e6-8a73-cff62d9268c5" },"@56c8d702f12e05ae64872e7c2a92073486b57623bf04941cef47bf2e8452ca7d":{ "UserName": "@56c8d702f12e05ae64872e7c2a92073486b57623bf04941cef47bf2e8452ca7d","NickName": "小桔同学","RemarkName": "" },"@4731ed48f9914e7b95ebec77803223863e2cd33497f8633c755d985a544c6922":{ "UserName": "@4731ed48f9914e7b95ebec77803223863e2cd33497f8633c755d985a544c6922","NickName": "wuli舞哩客服","RemarkName": "50fb16c0-ff09-11e6-9fe5-dda97284d25b" }}`)
+
+  const EXPECTED = {
+    id:           '@@f47ccb7762cf4dc7352508fb03b79b44cedd3d1e16152e7f1bae3d4cf7996eb2',
+    mentionId0:   '@56c8d702f12e05ae64872e7c2a92073486b57623bf04941cef47bf2e8452ca7d',
+    mentionId1:   '@4731ed48f9914e7b95ebec77803223863e2cd33497f8633c755d985a544c6922',
+  }
+
+  // Mock
+  const mockContactGetter = function (id) {
+    return new Promise((resolve, reject) => {
+      if (id !== EXPECTED.id && !(id in CONTACT_LIST)) return resolve({})
+      if (id === EXPECTED.id) {
+        setTimeout(() => {
+          return resolve(RAW_OBJ)
+        }, 10)
+      }
+      if (id in CONTACT_LIST) {
```

Comment:
To make mock room data, you need to return room rawObj for the room id:

```ts
if (id  === ROOM_ID) {
  resolve(ROOM_RAW_OBJ)
}
```

And all IDs of the room members should also be resolved to the right rawObj in mockContactGetter.